src/resizing.jl

```diff
@@ -125,32 +128,30 @@ end
 
 restrict_size(len::Integer) = isodd(len) ? (len+1)>>1 : (len>>1)+1
 
-## imresize
-function imresize!(resized, original::AbstractMatrix)
-    scale1 = (size(original,1)-1)/(size(resized,1)-0.999f0)
-    scale2 = (size(original,2)-1)/(size(resized,2)-0.999f0)
-    for jr = 0:size(resized,2)-1
-        jo = scale2*jr
-        ijo = trunc(Int, jo)
-        fjo = jo - oftype(jo, ijo)
-        @inbounds for ir = 0:size(resized,1)-1
-            io = scale1*ir
-            iio = trunc(Int, io)
-            fio = io - oftype(io, iio)
-            tmp = (1-fio)*((1-fjo)*original[iio+1,ijo+1] +
-                           fjo*original[iio+1,ijo+2]) +
-                  + fio*((1-fjo)*original[iio+2,ijo+1] +
-                         fjo*original[iio+2,ijo+2])
-            resized[ir+1,jr+1] = convertsafely(eltype(resized), tmp)
-        end
-    end
-    resized
+# imresize
+imresize(original::AbstractArray, dim1, dimN...) = imresize(original, (dim1,dimN...))
+
+function imresize{T,N}(original::AbstractArray{T,N}, short_size::NTuple)
+    len_short = length(short_size)
+    new_size = ntuple(i -> (i > len_short ? size(original,i) : short_size[i]), N)
+    imresize(original, new_size)
 end
 
-imresize(original, new_size) = size(original) == new_size ? copy!(similar(original), original) : imresize!(similar(original, new_size), original)
+function imresize{T,N}(original::AbstractArray{T,N}, new_size::NTuple{N})
+    if size(original) == new_size
+        copy!(similar(original), original)
+    else
+        imresize!(similar(original, new_size), original)
+    end
+end
 
-convertsafely{T<:AbstractFloat}(::Type{T}, val) = convert(T, val)
-convertsafely{T<:Integer}(::Type{T}, val::Integer) = convert(T, val)
-convertsafely{T<:Integer}(::Type{T}, val::AbstractFloat) = round(T, val)
-convertsafely{T}(::Type{T}, val) = convert(T, val)
+function imresize!{T,S,N}(resized::AbstractArray{T,N}, original::AbstractArray{S,N})
+    itp = interpolate(original, BSpline(Linear()), OnGrid())
```

Comment:
Might want to dispatch to a version of `imresize!` that accepts `original::AbstractInterpolation`. That way if people want higher-order interpolation they can get it easily.